ir/json_generator.h

```diff
@@ -98,6 +100,52 @@ class JSONGenerator {
         out << std::endl << --indent << "}";
     }
 
+    template<typename T>
+    void generate(const std::set<T> &v) {
+        out << "[" << std::endl;
+        if (v.size() > 0) {
+            auto it = v.begin();
+            out << ++indent;
+            generate(*it);
+            for (it++; it != v.end(); ++it) {
+                out << "," << std::endl << indent;
+                generate(*it);
+            }
+            out << std::endl << --indent;
+        }
+        out << "]";
+    }
```

Comment:
There's probably a way of generating these 3 identical functions as a single template function with an appropriate `std::enabble_if`, but I don't know what it is.

Reply:
:+1: :+1: :bangbang: